### PR TITLE
Add mdformat plugins

### PR DESCRIPTION
**Pull request by blink1073**
## References
Follow up to #13322, cf https://github.com/jupyterlab/jupyterlab/pull/13322#pullrequestreview-1160139441

## Code changes
Add some mdformat plugins to allow GitHub syntax.

## User-facing changes
None

## Backwards-incompatible changes
None

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -51,6 +51,8 @@ repos:
     rev: 0.7.16
     hooks:
       - id: mdformat
+        additional_dependencies:
+          [mdformat-gfm, mdformat-frontmatter, mdformat-footnote]
```
Comment (jtpio) [inline .pre-commit-config.yaml:55]: Wondering if it's possible to add these as `dev` dependencies.

With `prettier` it was possible to format markdown files automatically in IDEs like VS Code which was quite handy. I don't know if it's also done automatically with `mdformat` or if it needs additional IDE configuration?

Comment (blink1073) [reply .pre-commit-config.yaml:55]: Perhaps we should stick with Prettier for all but the changelog?

Comment (jtpio) [reply .pre-commit-config.yaml:55]: Yeah maybe. Anyway we can have a look separately.

**Review by jtpio — APPROVED** [review]
Thanks!

**Pull request merged**